File: tarea16/ultimaPalabra.php
<?php
    function ultimaPalabra($arreglo){
        $n=strlen($arreglo);
        $subArreglo=array();
        $k=0; #Iterador para subArreglo
        for($i=$n-1; $i>=0 ;$i--){
            if($arreglo[$i]!=" "){
                $subArreglo[$k]=$arreglo[$i];
                $k=$k+1;
            }else{
                break;
            }
        }
        $cad="";
        for($i=$k-1;$i>=0;$i--){
            $cad.=$subArreglo[$i];
        }
        return $cad;
    }
//Prueba:
$cadena="Hola Mundo";
$cadena2="Estados Unidos Mexicanos";

echo (ultimaPalabra($cadena));
echo "<br>";
echo (ultimaPalabra($cadena2));
?>

File: tarea16/ultimaPalabra.py
def ultimaPalabra(arreglo,n):
    subArreglo=[]
    k=0
    for i in range(n-1, -1,-1):
        
        if arreglo[i]!=" ":
            subArreglo.append(arreglo[i])
            k+=1
        else:
            break
    cad=""
    for i in range(k-1,-1,-1):
        cad+=subArreglo[i]
    
    return cad
cadena1="Hola Mundo"
cadena2="Estados Unidos Mexicanos"
print(ultimaPalabra(cadena1,len(cadena1)))
print(ultimaPalabra(cadena2,len(cadena2)))